src/ota/ota_handler.rs

```diff
@@ -54,368 +51,193 @@ pub struct OtaEvent {
     pub message: String,
 }
 
-struct OtaStatusMessage {
-    status_code: String,
-    message: String,
+pub(crate) struct OtaStatusMessage {
+    pub(crate) status_code: String,
+    pub(crate) message: String,
```

Comment:
Since the `ota_handler` is already private to the crate, you can remove the `pub(crate)` from struct and method declaration. 
```suggestion
pub struct OtaStatusMessage {
    pub status_code: String,
    pub message: String,
```